scripts/pyinstaller/make_exe.py

```diff
@@ -89,6 +94,25 @@ def clean_pyinstaller_build_dirs():
         print(f"Deleted build directory: {str(location)}")
 
 
+def zip_with_symlinks(source_dir: Path, output_zip: Path) -> None:
+    """Create a zip archive that preserves symlinks for unix systems"""
+    output_zip.parent.mkdir(parents=True, exist_ok=True)
+    with zipfile.ZipFile(output_zip, "w", zipfile.ZIP_DEFLATED) as zipf:
+        for root, dirs, files in os.walk(source_dir):
+            for name in dirs + files:
+                path = os.path.join(root, name)
+                arcname = os.path.relpath(path, source_dir)
+
+                if os.path.islink(path):
+                    info = zipfile.ZipInfo(arcname)
+                    stat_info = os.lstat(path)
+                    info.external_attr = stat_info.st_mode << 16
```

Comment:
Do we need to mask before bit shifting like this:

https://github.com/python/cpython/blob/9bb27cd1051e0e1ee2022fcaedb828643a84a30e/Lib/zipfile/__init__.py#L648
```suggestion
                    info.external_attr = (stat_info.st_mode & 0xFFFF) << 16
```

Reply:
From offline discussions, we realized that this would mask out the `S_IFLINK` bit (`0o120000`) which would defeat the purpose of this PR.

Reply:
correction it won't mask the file type bits. I've added it in to match cpython. Building the installer still shows the correct symlink + permissions.